host/crypto/openssl/rsa.c

```diff
@@ -490,35 +490,35 @@ oe_result_t oe_rsa_get_public_key_from_private(
 
     /* Get RSA private key */
     if (!(rsa_private = EVP_PKEY_get1_RSA(private_key_temp->pkey)))
-        OE_RAISE(OE_FAILURE);
+        OE_RAISE(OE_CRYPTO_ERROR);
 
     RSA_get0_key(rsa_private, &private_e, &private_n, NULL);
 
     /* Check if it's possible to get the public key. */
-    if (!private_n || !private_n)
-        OE_RAISE(OE_FAILURE);
+    if (!private_e || !private_n)
```

Comment:
Good catch. Is there some way for us to be confident that there are no such errors lurking around.
Especially in cryptographic code.
Would code-coverage be enough or do we need a more thorough testing strategy?

Reply:
Code coverage would be the strategy for general testing of implementation; in this example though, it's really a question of incorrect/inadequate return value validation and such things are usually caught through static analysis. I don't think there's a catch-all strategy here.

Reply:
Ok. It would be good if we can start investing in static-analysis on OE code base.